lib/takeaway.rb

```diff
@@ -0,0 +1,36 @@
+class Takeaway
+  attr_reader :menu_class, :order_class
+  def initialize(order_class = Order, menu_class = Menu,
```

Comment:
Nice use of dependency injection here.

Reply:
Interesting though that this class contains an instance of the `Menu` and an instance of `Order` which, in turn, also holds onto an instance of `Menu`. I wonder if you could remove this duplication be changing your object model a little?